tests/test_wire.py

```diff
@@ -263,13 +263,30 @@ def test_reset_value_as_string(self):
         self.assertEqual(r.reset_value, 1)
 
     def test_invalid_reset_value_too_large(self):
-        with self.assertRaisesRegex(pyrtl.PyrtlError, "cannot fit in the specified"):
+        with self.assertRaisesRegex(
+            pyrtl.PyrtlError,
+            "bitwidth specified \\(4\\) is insufficient to represent constant 16"
+        ):
             r = pyrtl.Register(4, reset_value=16)
 
     def test_invalid_reset_value_too_large_as_string(self):
         with self.assertRaisesRegex(pyrtl.PyrtlError, "cannot fit in the specified"):
             r = pyrtl.Register(4, reset_value="5'd16")
 
+    def test_negative_reset_value(self):
+        r = pyrtl.Register(4, reset_value=-4)
+        self.assertEqual(
+            pyrtl.helperfuncs.val_to_signed_integer(r.reset_value, r.bitwidth),
+            -4
+        )
+
+    def test_negative_reset_value_as_string(self):
+        r = pyrtl.Register(4, reset_value="-4'd1")
```

Comment:
Also add tests for when a Verilog-style `reset_value`'s bitwidth does not match the `Register`'s bitwidth